packages/opentelemetry-instrumentation/test/common/Instrumentation.test.ts

```diff
@@ -24,6 +25,23 @@ class TestInstrumentation extends InstrumentationBase {
   enable() {}
   disable() {}
   init() {}
+  _onRequire<T>(
+    module: InstrumentationModuleDefinition<T>,
+    exports: T,
+    name: string,
+    baseDir?: string
+  ): T {
+    return super._onRequire(module, exports, name, baseDir)
+  }
+  getModuleVersion(directory: string): string {
```

Comment:
not sure to understand why you re-implement them without modyfing the implementation ? Is it just to be apply to use `sinon.spy` ? (in this case i would prefer to have `// @ts-expect-error` instead without `protected` keywords on the base class

Reply:
Yeah this was simply for unit testing. The entire _requireOn method appears to be uncovered. I'll revisit the comments above and see if I can do this without exposing the existing private method.

Reply:
Refactored with private methods & @ts-expect-error